ark-plonk-book/src/chapter_1.md

```diff
@@ -0,0 +1,373 @@
+
+#### Point addition
+
+We will first compute $P+Q$ in the affine form and then projective form. 
+
+1. We have two points $P=(x_p,y_p)$ and $Q=(x_q,y_q)$ where $P\neq Q$. If $P=O$ then $P$ is the identity point which means $P+Q=Q$. Likewise if $Q=O$, then $P+Q=P$ \
+Else, $P+Q=S$ for $S=(x_s,y_s)$ where $S=-R$ (the point represented in the graph) such that: $$x_s=\lambda^2-x_p-x_q$$ and $$y_s=\lambda(x_p-x_s)-y_p$$   where      $\lambda=\dfrac{y_q-y_p}{x_q-x_p}$
+2. In the projective form, each elliptic curve point has 3 coordinates instead of 2 $(x,y,z)$ for $z\neq 0$ (for all points except the point at infinity). Using the projective form allows us to give coordinates to the point at infinity. It also speeds up some of the most used arithmetic operations in the curve. The forward mapping is given by $(x,y)\rightarrow (xz,yz,z)$ and reverse mapping is giving by $(x,y,z)(\dfrac{x}{z},\dfrac{y}{z})$. Let $P=(x_p,y_p,z_p)$ and $Q=(x_q,y_q,z_q)$ and $\dfrac{x_p}{z_q}\neq\dfrac{x_q}{z_p}$. 
+
+
+<figure>
+<img src="images/images/ECP.png"
+     alt="projective"
+     style="float: left; margin-right: 10px;" />
+<figcaption>Figure 2: Graphical representation of the map with z =1</figcaption>
+</figure>
+
+By expanding the previous arithmetics, we have:
+
+$\lambda=\dfrac{\dfrac{y_q}{z_q}-\dfrac{y_p}{z_p}}{\dfrac{x_q}{z_q}-\dfrac{x_p}{z_p}}=\dfrac{\dfrac{y_q}{z_q}-\dfrac{y_p}{z_p}}{\dfrac{x_q}{z_q}-\dfrac{x_p}{z_p}}*\dfrac{z_pz_q}{z_pz_q}$ 
+
+$\lambda=\dfrac{y_qz_p-y_pz_q}{x_qz_p-x_pz_q}$ 
+
+
+ $x_s=\lambda^2-\dfrac{x_p}{z_p}-\dfrac{x_q}{z_q}$ and     $y_s=\lambda(\dfrac{x_p}{z_p}-x_s)-\dfrac{y_p}{z_p}$
+
+#### Point doubling
+
+We will compute $P+P=2P$ in the affine form as follows:\
+If $P=O$ then $2P=O$ \
+Else  $P=(x,y)$:\
+    &emsp;&nbsp;If $y=0$ then $2P=O$  
+    &emsp;&nbsp;Else $2P=(x',y')$  
+
+such that: the derivative $\lambda=\dfrac{dy}{dx}=\dfrac{3x^2}{2y},\; x'=\lambda^2-2x\;$   and $y'=\lambda(x-x')-y=\lambda^3+3\lambda x-y$
+
+#### The discrete logarithm problem
+The security of many cryptographic techniques depends on the intractability of the discrete logarithm problem.
+
+#### Definition:
+Let $G$ be a multiplicative group. The discrete logarithm problem (DLP) is:
+Given $g,h\in G$ to find $a$, if it exists, such that $h=g^a$.
+
+
+#### Elliptic Curve Discrete Logarithm Problem  (ECDLP)
+
+Let $E$ be an elliptic curve of the Weierstrass form defined over a finite field $F_q$. Let $S$ and $T$ be two points in $E(F_q)$. Find an integer $m$ such that:
+$$T=mS$$
+
+The fastest method to solve the ECDLP problem in $E(F_q)$ is the [Pollard Rho](https://www.ams.org/mcom/1978-32-143/S0025-5718-1978-0491431-9/S0025-5718-1978-0491431-9.pdf) method which has exponential complexity $O(\sqrt{|G|})$.
```

Comment:
It would be instructive to say at some point earlier that we usually want to take elliptic curves over very large finite fields, `|F_p|≈2^256` so that this algorithm is actually "exponential".

Reply:
Yes, will add it :)